src/EditorFeatures/Core.Wpf/InheritanceMargin/InheritanceChainMarginTaggerProvider.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Implementation.Classification;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InheritanceMargin;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InheritanceMargin
+{
+    [Export(typeof(IViewTaggerProvider))]
+    [TagType(typeof(InheritanceMarginTag))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [Name(nameof(InheritanceChainMarginTaggerProvider))]
+    internal sealed class InheritanceChainMarginTaggerProvider : AsynchronousViewTaggerProvider<InheritanceMarginTag>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InheritanceChainMarginTaggerProvider(
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IForegroundNotificationService notificationService) : base(
+                threadingContext,
+                listenerProvider.GetListener(FeatureAttribute.InheritanceChainMargin),
+                notificationService)
+        {
+        }
+
+        protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+            => new CompilationAvailableTaggerEventSource(
+                subjectBuffer,
+                TaggerDelay.OnIdle,
+                ThreadingContext,
+                AsyncListener,
+                TaggerEventSources.OnWorkspaceChanged(subjectBuffer, TaggerDelay.OnIdle, AsyncListener),
+                TaggerEventSources.OnViewSpanChanged(ThreadingContext, textViewOpt, TaggerDelay.OnIdle, TaggerDelay.OnIdle),
+                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, TaggerDelay.OnIdle),
+                TaggerEventSources.OnOptionChanged(subjectBuffer, FeatureOnOffOptions.ShowInheritanceMargin, TaggerDelay.OnIdle));
```

Comment:
nit: consider setting the OnOptionCHanged to a shorter delay, so that if users enable/disalbe this, the change appears immediately.

Reply:
nit: probably a decent idea to doc why we're using CompilationAvailableTaggerEventSource